File: 4/java/tb.java
// Execute this testbench using the following command:
// javac -d . Solution.java; java tb.java

import medianOfTwoArrays.Solution;

class Testbench {
    public static void main(String[] args) {
        Solution s = new Solution();

        // test 1
        if (s.findMedianSortedArrays(new int[] {1}, new int[] {1}) != 1.0)
            throw new RuntimeException("FAILED TEST 1");

        // test 2
        if (s.findMedianSortedArrays(new int[] {1, 2}, new int[] {3}) != 2.0)
            throw new RuntimeException("FAILED TEST 2");

        // test 3
        if (s.findMedianSortedArrays(new int[] {1, 5}, new int[] {}) != 3.0)
            throw new RuntimeException("FAILED TEST 3");

        // test 4
        if (s.findMedianSortedArrays(new int[] {}, new int[] {3, 4, 5, 6, 7, 8}) != 5.5)
            throw new RuntimeException("FAILED TEST 4");

        // test 5
        if (s.findMedianSortedArrays(new int[] {1, 6, 7, 8, 9, 10}, new int[] {3, 5, 55, 665, 949}) != 8.0)
            throw new RuntimeException("FAILED TEST 5");

        System.out.println("PASSED ALL TESTS");
    }
}

File: 4/c++/tb.cpp
#include <iostream>
#include <vector>

using std::cout;
using std::endl;
using std::vector;

#include "solution.cpp"


int main() {
    Solution s;
    std::vector<int> nums1, nums2;

    // test 1
    nums1 = {1, 2, 3, 4, 5, 6, 7, 8, 9};
    nums2 = {};
    if (s.findMedianSortedArrays(nums1, nums2) != 5.0) {
        cout << "FAILED TEST 1" << endl;
        return 1;
    }
    // test 2
    nums1 = {1};
    nums2 = {2};
    if (s.findMedianSortedArrays(nums1, nums2) != 1.5) {
        cout << "FAILED TEST 2" << endl;
        return 1;
    }
    // test 3
    nums1 = {1, 4, 5};
    nums2 = {2, 3, 6};
    if (s.findMedianSortedArrays(nums1, nums2) != 3.5) {
        cout << "FAILED TEST 3" << endl;
        return 1;
    }
    // test 4
    nums1 = {1, 2, 3, 4, 5, 6};
    nums2 = {2};
    if (s.findMedianSortedArrays(nums1, nums2) != 3.0){
        cout << "FAILED TEST 4" << endl;
        return 1;
    }
    
    cout << "PASSED ALL TESTS" << endl;
    return 0;
}